packages/cli/src/constructs/check.ts

```diff
@@ -16,90 +16,215 @@ import { ConfigDefaultsGetter, makeConfigDefaultsGetter } from './check-config'
    * ap-south-1, af-south-1
+   * 
+   * @example ['us-east-1', 'eu-west-1', 'ap-southeast-1']
+   * @see {@link https://www.checklyhq.com/docs/monitoring/global-locations/ | Global Locations}
    */
   locations?: Array<keyof Region>
+  
   /**
    * An array of one or more private locations where to run the check.
    * PrivateLocation instances or slug name strings are allowed.
    *
    * `string` slug names are **only** allowed for private locations that **not** belong to the project. Use
    * PrivateLocation instances references for private locations created within the project.
+   * 
+   * @example
+   * ```typescript
+   * // Using private location instances
+   * privateLocations: [myPrivateLocation, anotherPrivateLocation]
+   * 
+   * // Using existing private location slugs
+   * privateLocations: ['my-datacenter-1', 'office-location']
+   * ```
    */
   privateLocations?: Array<string|PrivateLocation|PrivateLocationRef>
+  
   /**
-   * Tags for organizing and filtering checks.
+   * Tags for organizing and filtering checks in the dashboard.
+   * 
+   * @example ['production', 'api', 'critical']
    */
   tags?: Array<string>
+  
   /**
-   * How often the check should run in minutes.
+   * How often the check should run. Can be specified in minutes or using Frequency constants.
+   * 
+   * @example
+   * ```typescript
+   * frequency: Frequency.EVERY_5M  // Using frequency constant
+   * frequency: 10                   // Every 10 minutes
+   * ```
    */
   frequency?: number | Frequency
   /**
    * The id of the check group this check is part of. Set this by calling `someGroup.ref()`
    * @deprecated Use {@link group} instead.
    */
   groupId?: Ref
+  
   /**
    * The CheckGroup that this check is part of.
+   * Groups allow you to organize related checks and apply common settings.
+   * 
+   * @example
+   * ```typescript
+   * // Create a new check group
+   * const apiGroup = new CheckGroupV2('api-group', {
+   *   name: 'API Checks',
+   *   activated: true,
+   *   locations: ['us-east-1']
+   * })
+   * 
+   * // Reference an existing check group by ID
+   * const existingGroup = CheckGroupV2.fromId(123)
+   * 
+   * // Use in check configuration
+   * group: apiGroup  // or existingGroup
+   * ```
    */
   group?: CheckGroupV1 | CheckGroupV2 | CheckGroupRef
+  
   /**
    * List of alert channels to notify when the check fails or recovers.
    * If you don't set at least one, we won't be able to alert you in case something goes wrong with your check.
-   * @link {https://www.checklyhq.com/docs/alerting-and-retries/alert-channels/#alert-channels Alert channels}
+   * 
+   * @example
+   * ```typescript
+   * alertChannels: [
+   *   new EmailAlertChannel('email-alerts', { address: 'team@example.com' }),
+   *   new SlackAlertChannel('slack-alerts', { url: 'https://hooks.slack.com/...' })
+   * ]
+   * ```
```

Comment:
While this is a valid way to use alert channels, I think the example is a bit dangerous... should a user copy this as is, they'd probably end up creating a whole bunch of duplicate alert channels if they try to apply the same to multiple checks. And/or, get confused why they're getting `logicalId` conflicts. Instead, the example should emphasize storing the AlertChannel instances in variables, and then using the alert channels via those variable references.